### PR TITLE
MC-1120 upgrade java version to 17

### PR DESCRIPTION
**Pull request by zhmaeff**

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -1,7 +1,7 @@
-ARG MC_VERSION=5.1-BETA-1
+ARG MC_VERSION=5.1-SNAPSHOT
 ARG MC_INSTALL_NAME="hazelcast-management-center-${MC_VERSION}"
 ARG MC_INSTALL_JAR="hazelcast-management-center-${MC_VERSION}.jar"
-ARG MC_INSTALL_ZIP="${MC_INSTALL_NAME}.zip"
+ARG MC_INSTALL_ZIP="hazelcast-management-center-latest-snapshot.zip"
```
Comment (promanenko) [inline Dockerfile:4]: why is this line changed?

File: Dockerfile [hunk 2]
```diff
@@ -91,4 +92,4 @@ RUN echo "Adding non-root user" \
 USER ${USER_UID}
 
 # Start Management Center
-CMD ["bash", "./bin/mc-start.sh"]
+ENTRYPOINT ["bash", "./bin/mc-start.sh"]
```
Comment (promanenko) [inline Dockerfile:95]: isn't it going to break backward compatibility for cases like https://github.com/hazelcast/management-center-docker#configuring-management-center-inside-your-custom-docker-image  ?

File: Dockerfile (hunk 1 above)
Comment (zhmaeff) [reply Dockerfile:4]: Docker build was failing for master as there is no /bin/hz-mc in 5.1-BETA-1

File: Dockerfile (hunk 2 above)
Comment (zhmaeff) [reply Dockerfile]: rolled it back

File: Dockerfile (hunk 1 above)
Comment (zhmaeff) [reply Dockerfile:4]: And we don't publish `hazelcast-management-center-5.1-SNAPSHOT.zip`

Comment (promanenko) [reply Dockerfile:4]: > And we don't publish hazelcast-management-center-5.1-SNAPSHOT.zip
didn't see it

Comment (zhmaeff) [reply Dockerfile:4]: There is no package like `hazelcast-management-center-5.1-SNAPSHOT.zip` in `https://repository.hazelcast.com/download/management-center`
